Sim/SimG4Components/src/SimG4Svc.cpp

```diff
@@ -54,12 +59,20 @@ StatusCode SimG4Svc::initialize() {
   m_runManager.SetUserInitialization(m_detectorTool->detectorConstruction());
 
   G4UImanager* UImanager = G4UImanager::GetUIpointer();
-  for (auto command: m_g4PreInitCommands) {
+  for (auto command : m_g4PreInitCommands) {
     UImanager->ApplyCommand(command);
   }
 
-
   m_runManager.Initialize();
+  if (m_interactiveMode) {
+    G4VisManager* visManager = new G4VisExecutive;
+    visManager->Initialize();
+    // Define UI terminal for interactive mode
+    G4UIsession* session = new G4UIterminal;
+    session->SessionStart();
+    delete session;
```

Comment:
can't we work with smart pointers here, instead of bare ones?

Reply:
Hi Benedikt,

Where should I keep the additional markdown file?
I have changed the bare pointer solution into smart ones. The visManager, should be deleted at the end of the program, otherwise it causes a segmentation fault.

Thanks,
Nilou